app/uk/gov/hmrc/gform/sharedmodel/form/UserData.scala

```diff
@@ -16,16 +16,33 @@
 
 package uk.gov.hmrc.gform.sharedmodel.form
 
-import play.api.libs.json.{ Json, OFormat }
+import play.api.libs.json.{ Json, OFormat, OWrites, Reads }
+import uk.gov.hmrc.gform.sharedmodel.{ ListAllInfo }
+import play.api.libs.functional.syntax._
 
 case class UserData(
   formData: FormData,
   formStatus: FormStatus,
-  visitsIndex: VisitIndex
+  visitsIndex: VisitIndex,
+  obligations: Option[ListAllInfo]
```

Comment:
Why do we need an `Option`?

Reply:
Sometimes we don't have the information needed to make the API call at the start of the form. As such obligations would be None at that point

Reply:
But what is difference between `None` and empty `List` in that case?